packages/web-client/src/helpers/share/role.ts

```diff
@@ -359,26 +359,33 @@ export abstract class LinkShareRoles {
   static readonly all = [
     linkRoleViewerFile,
     linkRoleViewerFolder,
-    linkRoleContributorFolder,
     linkRoleEditorFolder,
     linkRoleUploaderFolder
   ]
 
   static list(
     isFolder: boolean,
     canEditFile = false,
+    canContribute = false,
     hasAliasLinks = false,
     hasPassword = false
   ): ShareRole[] {
     return [
       ...(hasAliasLinks && !hasPassword ? [linkRoleInternalFile, linkRoleInternalFolder] : []),
       ...this.all,
-      ...(canEditFile ? [linkRoleEditorFile] : [])
+      ...(canEditFile ? [linkRoleEditorFile] : []),
+      ...(canContribute ? [linkRoleContributorFolder] : [])
```

Comment:
With this change the `Contributor` role is at the end of the roles list. Should stay at the same position as before please.
I'd say at this point it doesn't make much sense to keep the `all` array at the top of the class. I'd like to propose that you inline all roles in the `list` function here to keep the correct order. And then in the `getByBitmask` function  just call `list(isFolder, true, true, true, false)` instead of building another array again.